fabtests/pytest/common.py

```diff
@@ -1,14 +1,49 @@
 import pytest
 import errno
+import os
+from tempfile import NamedTemporaryFile
+from subprocess import run
+from retrying import retry
 
+
+class SshConnectionError(Exception):
+
+    def __init__(self):
+        super().__init__(self, "Ssh connection failed")
+
+
+def is_ssh_connection_error(exception):
+    return isinstance(exception, SshConnectionError)
+
+
+def has_ssh_connection_err_msg(output):
+    err_msgs = ["kex_exchange_identification: Connection closed by remote host",
+                "ssh_exchange_identification: read: Connection reset by peer",
+                "port 22: Connection refused"]
+
+    for msg in err_msgs:
+        if output.find(msg) != -1:
+            return True
+
+    return False
```

Comment:
nit: I think this can be simplified to

```python3
return output in err_msgs
```

Reply:
That will not work because output is a large string, and the ssh error message is only part of it.

Reply:
I see. I was assuming `output` was the singular line/message.